tls_codec/derive/src/lib.rs

```diff
@@ -1,23 +1,80 @@
+//! Derive macros for traits in `tls_codec`
+//!
+//! The following attribute is available:
+//!
+//! ```text
+//! #[tls_codec(with = "module")]
+//! ```
+//! This attribute may be applied to a struct field. It indicates that deriving any of the
+//! `tls_codec` traits for the containing struct uses the following functions defined in the
+//! module `module`:
+//! - `tls_deserialize` when deriving `Deserialize`
+//! - `tls_serialize` when deriving `Serialize`
+//! - `tls_serialized_len` when deriving `Size`
+//!
+//! Their expected signatures match the corresponding methods in the traits.
+//!
+//! ```
+//! use tls_codec_derive::{TlsSerialize, TlsSize};
+//!
+//! #[derive(TlsSerialize, TlsSize)]
+//! struct Bytes {
+//!     #[tls_codec(with = "bytes")]
+//!     values: Vec<u8>,
+//! }
+//!
+//! mod bytes {
+//!     use std::io::Write;
+//!     use tls_codec::{Serialize, Size, TlsByteSliceU32};
+//!
+//!     pub fn tls_serialized_len(v: &[u8]) -> usize {
+//!         TlsByteSliceU32(v).tls_serialized_len()
+//!     }
+//!
+//!     pub fn tls_serialize<W: Write>(v: &[u8], writer: &mut W) -> Result<usize, tls_codec::Error> {
+//!         TlsByteSliceU32(v).tls_serialize(writer)
+//!     }
+//! }
+//! ```
+
 extern crate proc_macro;
 extern crate proc_macro2;
 
 use proc_macro::TokenStream;
 use proc_macro2::{Span, TokenStream as TokenStream2};
-use quote::{quote, ToTokens};
+use quote::quote;
 use syn::{
     self, parenthesized,
     parse::{ParseStream, Parser, Result},
-    parse_macro_input, Data, DeriveInput, Fields, FieldsNamed, FieldsUnnamed, Generics, Ident,
-    Index,
+    parse_macro_input, Attribute, Data, DeriveInput, ExprPath, Field, Generics, Ident, Index, Lit,
+    Member, Meta, NestedMeta, Type,
 };
 
+const ATTR_IDENT: &str = "tls_codec";
+
+#[derive(Clone)]
+enum Prefix {
+    Type(Type),
+    Custom(ExprPath),
+}
+
+impl Prefix {
+    fn for_trait(&self, trait_name: &str) -> TokenStream2 {
```

Comment:
I know this file doesn't have a lot of comments yet. But let's start adding some.

Reply:
I added more comments. Please let me know if that's ok. Thank you!